src/app/shared/services/auth.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, inject } from '@angular/core';
+import {
+  Auth,
+  User,
+  authState,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+} from '@angular/fire/auth';
+import { BehaviorSubject, Observable, from, map, tap } from 'rxjs';
+import { IAuthCredentials } from '../models/auth-credentials.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  private auth = inject(Auth);
+
+  private userSubject = new BehaviorSubject<User | null>(null);
```

Comment:
try to use signal instead of subject. I'm not sure it would work, but you could check it.
here is a reference
https://modernangular.com/articles/service-with-a-signal-in-angular